src/lib/assets/assets.ts

```diff
@@ -0,0 +1,169 @@
+import { helmLayout, createWebhookYaml, toYaml } from "./index";
+import { loadCapabilities } from "./loader";
+import { namespaceComplianceValidator, dedent } from "../helpers";
+import { storeRole, storeRoleBinding, clusterRoleBinding, serviceAccount } from "./rbac";
+import { watcherService, service, tlsSecret, apiTokenSecret } from "./networking";
+import { webhookConfig } from "./webhooks";
+import { generateZarfYaml, generateZarfYamlChart, generateAllYaml, overridesFile } from "./yaml";
+import { promises as fs } from "fs";
+
+export class Assets {
+  readonly name: string;
+  readonly tls: TLSOut;
+  readonly apiToken: string;
+  readonly alwaysIgnore!: WebhookIgnore;
+  capabilities!: CapabilityExport[];
+
+  image: string;
+  buildTimestamp: string;
+  hash: string;
+
+  constructor(
+    readonly config: ModuleConfig,
+    readonly path: string,
+    readonly host?: string,
+  ) {
+    this.name = `pepr-${config.uuid}`;
+    this.buildTimestamp = `${Date.now()}`;
+    this.alwaysIgnore = config.alwaysIgnore;
+    this.image = `ghcr.io/defenseunicorns/pepr/controller:v${config.peprVersion}`;
+    this.hash = "";
+    // Generate the ephemeral tls things
+    this.tls = genTLS(this.host || `${this.name}.pepr-system.svc`);
+
+    // Generate the api token for the controller / webhook
+    this.apiToken = crypto.randomBytes(32).toString("hex");
+  }
+
+  // TODO: Appears unused
+  setHash = (hash: string): void => {
+    this.hash = hash;
+  };
+
+  deploy = async (force: boolean, webhookTimeout?: number): Promise<void> => {
+    this.capabilities = await loadCapabilities(this.path);
+    await deploy(this, force, webhookTimeout);
+  };
+
+  zarfYaml = (path: string): string => generateZarfYaml(this.name, this.image, this.config, path);
+
+  zarfYamlChart = (path: string): string => generateZarfYamlChart(this.name, this.image, this.config, path);
+
+  allYaml = async (imagePullSecret?: string): Promise<string> => {
+    this.capabilities = await loadCapabilities(this.path);
+    // give error if namespaces are not respected
+    for (const capability of this.capabilities) {
+      namespaceComplianceValidator(capability, this.alwaysIgnore?.namespaces);
+    }
+
+    const mutateWebhook = await webhookConfig(this, "mutate", this.config.webhookTimeout);
+    const validateWebhook = await webhookConfig(this, "validate", this.config.webhookTimeout);
+
+    const code = await fs.readFile(this.path);
+
+    // Generate a hash of the code
+    this.hash = crypto.createHash("sha256").update(code).digest("hex");
+
+    const watchDeployment = getWatcher(this, this.hash, this.buildTimestamp, imagePullSecret);
+    const deployment = getDeployment(this, this.hash, this.buildTimestamp, imagePullSecret);
+
+    const assetsInputs = {
+      apiToken: this.apiToken,
+      capabilities: this.capabilities,
+      config: this.config,
+      hash: this.hash,
+      name: this.name,
+      path: this.path,
+      tls: this.tls,
+    };
```

Comment:
Placeholder object that contains relevant properties from assets class. Consider alternatives.